app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsItemType.kt

```diff
@@ -0,0 +1,152 @@
+import android.view.View
+import com.xwray.groupie.kotlinandroidextensions.Item
+import com.xwray.groupie.kotlinandroidextensions.ViewHolder
+import io.reactivex.subjects.Subject
+import kotlinx.android.synthetic.main.list_patient_search.*
+import kotlinx.android.synthetic.main.list_patient_search_header.*
+import org.simple.clinic.R
+import org.simple.clinic.facility.Facility
+import org.simple.clinic.patient.PatientSearchResult
+import org.simple.clinic.util.estimateCurrentAge
+import org.simple.clinic.widgets.UiEvent
+import org.threeten.bp.Clock
+import org.threeten.bp.ZoneOffset
+import org.threeten.bp.format.DateTimeFormatter
+
+sealed class PatientSearchResultsItemType : Item() {
+
+  data class Header(
+      private val stringResId: Int,
+      private val arguments: List<String> = emptyList()
+  ) : PatientSearchResultsItemType() {
+
+    override fun getLayout(): Int = R.layout.list_patient_search_header
+
+    override fun bind(viewHolder: ViewHolder, position: Int) {
+      val header = viewHolder.patientsearch_header
+
+      header.text = viewHolder.itemView.context.getString(stringResId, *arguments.toTypedArray())
+    }
+  }
+
+  object NoPatients : PatientSearchResultsItemType() {
+    override fun getLayout(): Int = R.layout.list_patient_search_no_patients
+
+    override fun bind(viewHolder: ViewHolder, position: Int) {
+    }
+  }
+
+  data class Patient(
+      private val searchResult: PatientSearchResult,
+      private val currentFacility: Facility,
+      private val itemClicks: Subject<UiEvent>,
+      private val phoneObfuscator: PhoneNumberObfuscator,
+      private val dateOfBirthFormat: DateTimeFormatter,
+      private val clock: Clock
+  ) : PatientSearchResultsItemType() {
+
+    override fun getLayout(): Int = R.layout.list_patient_search
+
+    override fun bind(viewHolder: ViewHolder, position: Int) {
+      val genderImageView = viewHolder.patientsearchresult_item_gender
+      val titleTextView = viewHolder.patientsearchresult_item_title
+      val addressTextView = viewHolder.patientsearchresult_item_address
+      val dateOfBirthTextView = viewHolder.patientsearchresult_item_dateofbirth
+      val phoneNumberTextView = viewHolder.patientsearchresult_item_phone
+      val lastBpDateAndFacilityTextView = viewHolder.patientsearchresults_item_last_bp
+      val lastBpDateFrame = viewHolder.patientsearchresults_item_last_bp_container
+
+      viewHolder.itemView.setOnClickListener {
+        itemClicks.onNext(PatientSearchResultClicked(searchResult))
+      }
+
+      fun renderLastRecordedBloodPressure(
+          dateOfBirthFormat: DateTimeFormatter,
+          currentFacility: Facility,
+          resources: Resources
+      ) {
+        val lastBp = searchResult.lastBp
+        if (lastBp == null) {
+          lastBpDateFrame.visibility = View.GONE
+        } else {
+          lastBpDateFrame.visibility = View.VISIBLE
+
+          val lastBpDate = lastBp.takenOn.atZone(ZoneOffset.UTC).toLocalDate()
```

Comment:
This will print the time in UTC, which is wrong. We'll need to provide the `UserClock` and read the zone from it and convert it into the local time like so:

```kotlin
val lastBpDate = lastBp
            .takenOn
            .atZone(ZoneOffset.UTC)
            .withZoneSameInstant(userClock.zone)
            .toLocalDate()
```

Reply:
done